packages/web3-net/test/unit/rpc_methods.test.ts

```diff
@@ -0,0 +1,44 @@
+import { Web3RequestManager } from 'web3-core';
+
+import { getId, getPeerCount, isListening } from '../../src/rpc_methods';
+
+describe('rpc_methods', () => {
+	const requestManagerSendSpy = jest.fn();
+
+	let requestManager: Web3RequestManager;
+
+	beforeAll(() => {
+		requestManager = new Web3RequestManager();
+		requestManager.setProvider('http://127.0.0.1:8545');
+		requestManager.send = requestManagerSendSpy;
```

Comment:
Try not to override any property during the tests.

```suggestion
		requestManagerSendSpy = jest.spyOn(requestManager, 'send');
```

https://jestjs.io/docs/jest-object#jestspyonobject-methodname

Reply:
for some `unknown` reason to me using spy result with an error

![image](https://user-images.githubusercontent.com/6376047/155557452-820f7f16-bf36-4413-93f6-ea9b32d14c71.png)

maybe you have some advice to do a workaround?

Reply:
As these tests are just verifying that the `send` method from the `requestManager` is receiving the expected JSON RPC params, I believe how the code is written is okay. It's also how it's being done in
[web3-eth/test/unit/rpc_methods_with_parameters.test.ts](https://github.com/ChainSafe/web3.js/blob/ea41c556130558b37305767a39f25b4306dc0e50/packages/web3-eth/test/unit/rpc_methods_with_parameters.test.ts#L86)  to avoid actually making a request to the provider

Reply:
> As these tests are just verifying that the `send` method from the `requestManager` is receiving the expected JSON RPC params, I believe how the code is written is okay. It's also how it's being done in [web3-eth/test/unit/rpc_methods_with_parameters.test.ts](https://github.com/ChainSafe/web3.js/blob/ea41c556130558b37305767a39f25b4306dc0e50/packages/web3-eth/test/unit/rpc_methods_with_parameters.test.ts#L86) to avoid actually making a request to the provider

already did as is in web3-eth